sonarqube/httpHelpers.go

```diff
@@ -57,3 +60,13 @@ func httpRequestHelper(client *retryablehttp.Client, method string, sonarqubeURL
 
 	return *resp, nil
 }
+
+// https://github.com/jdamata/terraform-provider-sonarqube/issues/201
+// go-retryablehttp error contains the token/user:pass in plaintext.
+// We want to censor that secret before logging the error
+func censorError(err error, secret string) error {
+	// Replace the secret with asterisks of the same length
+	asterisks := strings.Repeat("*", len(secret))
+	result := strings.Replace(err.Error(), secret, asterisks, -1)
```

Comment:
The problem with this (admittedly a small one) is that you are still giving away information about the secret - ie the length of it because you return the same number of characters as the original.
Why not simply replace with say four asteriaks - `****` - in all cases so that you give nothing away?